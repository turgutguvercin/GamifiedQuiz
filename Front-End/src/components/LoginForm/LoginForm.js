import React, { useState } from "react";
import axios from "axios";
import { useNavigate } from "react-router-dom";
import './LoginForm.css';

function LoginForm(){
    let navigate = useNavigate();
    const [data, setData] = useState({
      email:"",
      password:"" 
    })

    const handleChange =(e)=> {
        setData({...data,[e.target.name]:e.target.value});
    } 

    const submitForm=(e)=> {
        e.preventDefault();
        const sendData = {
            email:data.email,
            password:data.password
        }

        axios.post('http://localhost/signInProcess.php', sendData, {withCredentials: true})
            .then((result) => {
                if(result.data.UserID === undefined || result.data.UserID === "notFound"){
                    alert("We couldn't find an account that matches what you entered")
                    console.log(result.data.UserID);
                }
                else{
                    navigate("../dashboard");
                    console.log(result.data.UserID);
                }
            })
            .catch((error) => {
                console.error("An error occurred:", error);
                alert("An error occurred while trying to log in.");
            })
    }

    return(
        <div className="loginform">
            <form onSubmit={submitForm} >
            <h3>Log In</h3>
            <br/>
            <label>Email</label>
            <input data-testid="test-input-email" type="email" name="email" className="input" onChange={handleChange} value={data.email} required />
            <br/>
            <label>Password</label>
            <input data-testid="test-input-password" type="password" name="password"  className="input" onChange={handleChange} value={data.password}  required />
            <br/>
        
            <input data-testid="test-input-button" type="submit" className="submit" name="submit" value="Login" />
            </form>
            <br/>
            <div className="bottomnotes">
            <p>By signing in you agree to <a className="terms" href="#">terms & conditions</a>.</p>
            <hr/>
            <p>Don't have an account?<a className="terms" href="../signup"> Click here to sign up.</a></p>
            </div>
        </div>
    );
}

export default LoginForm;
