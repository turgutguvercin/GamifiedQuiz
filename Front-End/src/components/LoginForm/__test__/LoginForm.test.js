import {render, screen} from '@testing-library/react';
import { BrowserRouter} from 'react-router-dom';
import LoginForm from '../LoginForm';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
const MockLoginForm = () => {
return(
<BrowserRouter>
<LoginForm />
</BrowserRouter>
);
}

describe("test login form component", () =>{

test('should render input element correctly', async () => {
    render(<MockLoginForm />);
    const inputElement = screen.getByTestId('test-input-email');
    userEvent.type(inputElement, "test@mail.com");
    expect(inputElement).toHaveValue("test@mail.com"); 
})

test('should submit input element correctly', async () => {
    render(<MockLoginForm />);
    const inputElementEmail = screen.getByTestId('test-input-email');
    userEvent.type(inputElementEmail, "test@mail.com");
    const inputElementPassword = screen.getByTestId('test-input-password');
    userEvent.type(inputElementPassword, "testpassword");
    userEvent.click(screen.getByTestId('test-input-button', {name: /submit/i}))

})
})


it('renders correctly', () => {
  const tree = renderer
    .create(<MockLoginForm/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});