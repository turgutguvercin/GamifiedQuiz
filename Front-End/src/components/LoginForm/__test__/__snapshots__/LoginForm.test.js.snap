// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="loginform"
>
  <form
    onSubmit={[Function]}
  >
    <h3>
      Log In
    </h3>
    <br />
    <label>
      Email
    </label>
    <input
      className="input"
      data-testid="test-input-email"
      name="email"
      onChange={[Function]}
      required={true}
      type="email"
      value=""
    />
    <br />
    <label>
      Password
    </label>
    <input
      className="input"
      data-testid="test-input-password"
      name="password"
      onChange={[Function]}
      required={true}
      type="password"
      value=""
    />
    <br />
    <input
      className="submit"
      data-testid="test-input-button"
      name="submit"
      type="submit"
      value="Login"
    />
  </form>
  <br />
  <div
    className="bottomnotes"
  >
    <p>
      By signing in you agree  to 
      <a
        className="terms"
        href="#"
      >
        terms & conditions
      </a>
      .
    </p>
    <hr />
    <p>
      Don't have an account?
      <a
        className="terms"
        href="#"
      >
         Click here to sign up.
      </a>
    </p>
  </div>
</div>
`;
